packages/arb-token-bridge-ui/src/hooks/useNetworks.ts

```diff
@@ -0,0 +1,255 @@
+import { Chain } from 'wagmi'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { useCallback, useMemo } from 'react'
+import { mainnet, arbitrum, goerli, arbitrumGoerli } from '@wagmi/core/chains'
+
+import { useArbQueryParams } from './useArbQueryParams'
+import {
+  ChainId,
+  getCustomChainsFromLocalStorage,
+  rpcURLs
+} from '../util/networks'
+import {
+  sepolia,
+  arbitrumNova,
+  arbitrumSepolia,
+  xaiTestnet,
+  stylusTestnet,
+  localL1Network as local,
+  localL2Network as arbitrumLocal
+} from '../util/wagmi/wagmiAdditionalNetworks'
+
+import { getPartnerChainsForChainId } from '../util/wagmi/getPartnerChainsForChainId'
+
+function getChainByChainId(chainId: ChainId): Chain {
+  const chain = {
+    // L1
+    [mainnet.id]: mainnet,
+    // L1 Testnet
+    [goerli.id]: goerli,
+    [sepolia.id]: sepolia,
+    // L2
+    [arbitrum.id]: arbitrum,
+    [arbitrumNova.id]: arbitrumNova,
+    // L2 Testnet
+    [arbitrumGoerli.id]: arbitrumGoerli,
+    [arbitrumSepolia.id]: arbitrumSepolia,
+    // L3
+    [xaiTestnet.id]: xaiTestnet,
+    [stylusTestnet.id]: stylusTestnet,
+    // E2E
+    [local.id]: local,
+    [arbitrumLocal.id]: arbitrumLocal
+  }[chainId]
+
+  return chain ?? mainnet
```

Comment:
Should we default here or just throw? We should probably do sanitization prior to calling this function

Reply:
I have made some changes, it's only called in `useNetworks` now, after sanitization. I have updated type, so we don't need to default to anything anymore